src/stitch.rs

```diff
@@ -97,6 +102,20 @@ where T: Clone,
     Ok((sorted_vec, offset_sorted))
 }
 
+fn check_file_format(name: &Path) -> Result<FileFormat> {
+    let ext = match name.extension() {
+        Some(e) => e.to_str().unwrap(),
+        None => return Err(ScalpelError::UnknownFileFormat.context(format!("No extension found") ).into()),
+    };
+
+    match ext {
```

Comment:
Maybe impl the `trait FromStr` directly for `FileFormat` or use a impl of `from_file_extension`?

Reply:
Also, for the intermediate future, see #4

Reply:
I think I'll go for the `FromStr`, but if #4 is handling this stuff anyways, I'd leave it for now?